src/Element/ExecutablePluginSelect.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Drupal\commerce\Element;
+
+use Drupal\Component\Utility\Html;
+use Drupal\Component\Utility\NestedArray;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\PluginFormInterface;
+use Drupal\Core\Render\Element\FormElement;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @FormElement("commerce_executable_plugin_select")
+ */
+class ExecutablePluginSelect extends FormElement {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getInfo() {
+    $class = get_class($this);
+    return [
+      '#input' => TRUE,
+      '#plugin_type_id' => NULL,
+      '#process' => [
+        [$class, 'processExecutablePluginSelect'],
+        [$class, 'processAjaxForm'],
+      ],
+    ];
+  }
+
+  /**
+   * Process callback.
+   */
+  public static function processExecutablePluginSelect(&$element, FormStateInterface $form_state, &$complete_form) {
+    if (!$element['#plugin_type_id']) {
+      throw new \InvalidArgumentException('You must specify the plugin type ID.');
+    }
+
+    $element['#tree'] = TRUE;
+
+    /** @var \Drupal\Core\Executable\ExecutableManagerInterface $executable_manager */
+    $executable_manager = \Drupal::service('plugin.manager.' . $element['#plugin_type_id']);
+
+    $values = $element['#value'];
+
+    $executable_id = !empty($values['executable_id']) ? $values['executable_id'] : '_none';
```

Comment:
Executable id sounds confusing. Sounds like a bit of verbosity would help. $executable_plugin_id?